### ISSUE TITLE
pyarrow.lib.ArrowInvalid: cannot mix struct and non-struct, non-null values

### ISSUE THREAD

**Issue by chrisrana** [awslabs/aws-data-wrangler#463]
I have table my_table with following schema in postgressql. I am trying to convert table data in to parquet. I am getting error.

 id                   | bigint
 key                  | character varying
 title                | character varying
 data                 | jsonb                                 

I understand that this error is because of jsonb field (data)...
Anyway to handle this

```
dfs = wr.db.read_sql_query("select id, data,key from my_table", con=engine,chunksize=700)
for df in dfs:
    if not df.empty:
                    wr.s3.to_parquet(df=df,path=db_s3_bucket_map[db_name],dataset=True,sanitize_columns=False )
```

Error:

```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<stdin>", line 1, in <listcomp>
  File "/usr/local/lib/python3.7/site-packages/awswrangler/db.py", line 234, in _iterate_cursor
    yield _records2df(records=records, cols_names=cols_names, index=index, safe=safe, dtype=dtype)
  File "/usr/local/lib/python3.7/site-packages/awswrangler/db.py", line 199, in _records2df
    array = _data_types.process_not_inferred_array(ex, values=col_values)  # Creating Arrow array
  File "/usr/local/lib/python3.7/site-packages/awswrangler/_data_types.py", line 358, in process_not_inferred_array
    dtype = process_not_inferred_dtype(ex=ex)
  File "/usr/local/lib/python3.7/site-packages/awswrangler/_data_types.py", line 343, in process_not_inferred_dtype
    raise ex
  File "/usr/local/lib/python3.7/site-packages/awswrangler/db.py", line 197, in _records2df
    array: pa.Array = pa.array(obj=col_values, safe=safe)  # Creating Arrow array
  File "pyarrow/array.pxi", line 296, in pyarrow.lib.array
  File "pyarrow/array.pxi", line 39, in pyarrow.lib._sequence_to_array
  File "pyarrow/error.pxi", line 122, in pyarrow.lib.pyarrow_internal_check_status
  File "pyarrow/error.pxi", line 84, in pyarrow.lib.check_status
pyarrow.lib.ArrowInvalid: cannot mix struct and non-struct, non-null values
```

**Comment by igorborgest** [comment]
Unfortunately this is a limitation inherited from Apache Arrow, we will keep it blocked by now.

**Comment by igorborgest** [comment]
This will be naturally supported after Apache Arrow overcome this limitation.

**Comment by cristianmtr** [comment]
Any update on this? Any possible workaround?

**Comment by MattiasMTS** [comment]
I came across this and solved it by specifying the json dtype column (`data`column in your example`) as "string" solved it for me (awswrangler version: 2.17.0). However, this was not for Postgres, so it might not solve it regardless. Worth a shot at least!

**Comment by mausam-digital** [comment]
Any update on it yet?
